synapse/federation/federation_base.py

```diff
@@ -144,7 +143,7 @@ def callback(_, pdu, redacted):
                     )
                     return redacted
 
-                if spamcheck.check_event_for_spam(pdu):
```

Comment:
So we're requiring this to be run synchronously? I.e., no IO while checking?

Reply:
yeah, same as before basically (it doesn't yield on the function call)

Reply:
I was more wondering whether we really want to preclude the spam checkers from e.g. accessing the DB, but ok.